### PR TITLE
added 'project' module integration test

### PR DESCRIPTION
**Pull request by sarathbrp**

### CODE REVIEW

**Comment by kdelee** [comment]
@Alex-Izquierdo can you review for merge to `api-module` branch -- idea is @sarathbrp would incrementally add coverage for each module, then we can figure out where to run these tests.

File: tests/integration/project/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    description: "{{ description }}"
+    url: "{{ scm_url }}"
+    credential: "{{ credential_name }}"
+    state: present
+  register: project_creation
+
+- name: Assert that the project was created successfully
+  assert:
+    that:
+      - project_creation is changed
```
Comment (Alex-Izquierdo) [inline tests/integration/project/tasks/main.yml:40]: Maybe this assertion should be performed against the response of the api using the uri module.  The module could return changed and success without creating the project due to a bug.  Applicable to all assertions.

**Review by Alex-Izquierdo — COMMENTED** [review]
Some suggestions and questions, but it LGTM

**Comment by Alex-Izquierdo** [comment]
Adding QE staff as reviewers as this is of their interest.

**Review by dhaustein — APPROVED** [review]

File: tests/integration/project/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
+# This YAML file contains Ansible tasks for managing projects.
+# The tasks include generating a test ID, creating a project, updating the project, and deleting the project.
+- name: Define variables for credential and project
+  set_fact:
+    credential_name: "Test_Credential_{{ 1000000 | random }}"
+    project_name: "Test_Project_{{ 1000000 | random }}"
+    description: "Test Project Description"
+    scm_url: "https://github.com/your-repository/your-project.git"
```
[the remark this answers is not in the document]
Comment (sarathbrp) [reply tests/integration/project/tasks/main.yml:9]: create a var by default it uses the project you mentioned in the comment.

File: tests/integration/project/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    that:
+      - credential_creation is changed
+      - credential_creation is success
+
+- name: Create a new project
+  project:
+    name: "{{ project_name }}"
+    description: "{{ description }}"
+    url: "{{ scm_url }}"
+    credential: "{{ credential_name }}"
```
[the remark this answers is not in the document]
Comment (sarathbrp) [reply tests/integration/project/tasks/main.yml:33]: The credential_creation return the following:

    "credential_creation": {
        "ansible_facts": {
            "discovered_interpreter_python": "/usr/bin/python3"
        },
        "changed": true,
        "failed": false,
        "id": 15
    }

I think for project to create name of the credential should be good enough. Let me know if I miss understood your comment.

**Comment by sarathbrp** [comment]
@Alex-Izquierdo  I have added other modules as well, please go through it and let me know your comments.

File: tests/integration/credential/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  when: random_string is not defined
+
+- name: Generate a ID for the test
+  set_fact:
+    test_id: "{{ random_string | to_uuid }}"
+  when: test_id is not defined
+
+- name: configure scm_url
+  set_fact:
+    scm_url: "https://github.com/Alex-Izquierdo/eda-sample-project"
```
Comment (ttuffin) [inline tests/integration/credential/tasks/main.yml:17]: We just migrated this repo to ansible org: https://github.com/ansible/eda-sample-project

I won't block this PR, but please update this in a follow up PR.

**Pull request merged**